lib/action/bottom_modal_action.dart

```diff
@@ -100,12 +96,6 @@ class DismissBottomModalAction extends EnsembleAction {
   @override
   Future<dynamic> execute(BuildContext context, ScopeManager scopeManager,
       {DataContext? dataContext}) {
-    BuildContext? bottomModalContext =
-        ContextScopeWidget.getRootContext(context);
-    if (bottomModalContext != null) {
-      return Navigator.maybePop(
-          bottomModalContext, scopeManager.dataContext.eval(payload));
-    }
-    return Future.value(null);
+    return Navigator.maybePop(context, scopeManager.dataContext.eval(payload));
```

Comment:
so this will only work for the simplest use case where the widget is defined directly in the same block. The context here can belong to a deeply nested widget - hence the reason we want to travel up to the root to dismiss the dialog.

Reply:
Okay @vusters, actually `ContextScopeWidget.getRootContext(context)` was returning null, so the dialog was never dismissed.

Reply:
@vusters I've done this, if the `rootContext` is not null, it will use root context, else it will use the local context to dismiss the modal

Reply:
and this works even when called from deeply nested widget?